internal/controller/linstorsatellite_controller.go

```diff
@@ -259,6 +259,30 @@ func (r *LinstorSatelliteReconciler) kustomizeNodeResources(ctx context.Context,
 		return nil, err
 	}
 
+	if lsatellite.Spec.ClusterRef.ExternalController != nil {
+		lc, err := linstorhelper.NewClientForCluster(
```

Comment:
Create a helper for the helper? There are several call sites that could be reduced to `LinstorSatelliteReconciler.NewClientForCluster(ctx, lsatellite)`.

The `r *LinstorNodeConnectionReconciler` case is probably too complex to unify with the `LinstorSatelliteReconciler` ones.

Reply:
Did a half-measure, as I usually dislike having helpers that just call another function "in a special way". Downside is that we now need to construct a "ClusterReference" in the LinstorCluster case, but that seemed reasonable to me.